scripts/pr_tests.py

```diff
@@ -16,8 +16,37 @@
 
 import argparse
 import sys
+import os
+import re
 from github import Github
 
+
+def get_script_dependencies(script_path, folder_prefix):
+    dependencies = set()
+    with open(script_path, 'r') as script_file:
+        script_content = script_file.read()
+
+        # Using regular expression to find 'source' followed by a script path
+        pattern = r'^source\s+["\']?(.+?)["\']?$'
+        matches = (match for match in re.findall(pattern, script_content, re.MULTILINE) if match.startswith(folder_prefix))
+        dependencies.update(matches)
+
+    return dependencies
+
+
+def get_script_dependencies_graph(folder_prefix):
+    matching_files = [os.path.join(root, file_name) for root, _, files in os.walk(folder_prefix) for file_name in files if file_name.endswith(".sh")]
```

Comment:
Only need to check *test.sh because snips.sh files never source other ones.
```suggestion
    matching_files = [os.path.join(root, file_name) for root, _, files in os.walk(folder_prefix) for file_name in files if file_name.endswith("test.sh")]
```

Reply:
Ah good to know, I see that snips.sh only imports the .sh from the boilerplate folder whose changes will anyway trigger a complete test-rerun thanks to
https://github.com/istio/istio.io/blob/568245276f54cc9da3861a3bffe1bed64e1419aa/scripts/pr_tests.py#L95-L100
Changed to filter to *test.sh only files.